### PR TITLE
Integrate Makefile into automation

### PR DESCRIPTION
**Pull request by glower**
* Integrated Makefile into automation processes
* Updated documentation about Makefile usage
* Added documentation about LXD build
* Added missing Makefile targets for frontend tests and for code vendoring

### CODE REVIEW

**Comment by glower** [comment]
I added two new github actions to run frontend tests and backend tests. Currently we use TravisCI for this tasks. I would suggest to deactivate TravisCI integration. In my experience github actions are much faster and more reliable.

File: run-checks [hunk 1]
```diff
@@ -166,13 +164,10 @@ if [ ! -z "$UNIT" ]; then
     mkdir .coverage
     echo "mode: $COVERMODE" > .coverage/coverage.out
 
-    echo Building
```
Comment (glower) [inline run-checks:169]: we are building in a separate action, so don't need to do this here

File: README.md [hunk 2]
```diff
@@ -29,20 +31,46 @@ in order to changes take effect. That could require a browser restart.
 NEVER set this configuration in production environments.
 
 ## Install from Source
-If you have a Go development environment set up, Go get it, we recommend at least Go v1.13 or higher.
+
+If you have a Go development environment set up, we recommend at least Go v1.13 or higher. This project is build
```
Comment (doismellburning) [inline README.md:35]: s/build/built/

File: README.md [hunk 3]
```diff
@@ -29,20 +31,46 @@ in order to changes take effect. That could require a browser restart.
...
+
+```bash
+@serial-vault-focal:~/serial-vault$ tree bin/
+bin/
+├── factory
+├── serial-vault
+├── serial-vault-admin
+```
+
+To run Serial Vault in admin/UI mode you will need `static/` folder. Make sure to setup
```
Comment (doismellburning) [inline README.md:61]: s/need/need a/

File: README.md [hunk 4]
```diff
@@ -29,20 +31,46 @@ in order to changes take effect. That could require a browser restart.
...
+```bash
+@serial-vault-focal:~/serial-vault$ tree bin/
+bin/
+├── factory
+├── serial-vault
+├── serial-vault-admin
+```
+
+To run Serial Vault in admin/UI mode you will need `static/` folder. Make sure to setup 
+correct path to this folder in the configuration file with `docRoot` variable.
```
Comment (doismellburning) [inline README.md:62]: s/with/with the/

File: README.md [hunk 5]
```diff
@@ -112,10 +142,13 @@ To remove a dependency
 - Run `go mod tidy` in the source folder to remove dependency from go.mod file.
 
 ### Install Go
-Follow the instructions to [install Go](https://golang.org/doc/install).
+
+Follow the instructions to [install Go](https://golang.org/doc/install). The current version of the service runns with `Go 1.13`.
```
Comment (doismellburning) [inline README.md:146]: s/runns/runs/

File: README.md [hunk 6]
```diff
@@ -29,20 +31,46 @@ in order to changes take effect. That could require a browser restart.
 NEVER set this configuration in production environments.
 
 ## Install from Source
-If you have a Go development environment set up, Go get it, we recommend at least Go v1.13 or higher.
+
+If you have a Go development environment set up, we recommend at least Go v1.13 or higher. This project is build 
+and tested on Ubuntu 20.04.1 LTS` (Focal Fossa) with `go 1.13`. If you with to build or run the service from source we
```
Comment (doismellburning) [inline README.md:36]: s/with/wish/

File: README.md [hunk 7]
```diff
@@ -29,20 +31,46 @@ in order to changes take effect. That could require a browser restart.
...
+recommend using LXD contaner for this purpose. To get started with LXD, please follow this
```
Comment (doismellburning) [inline README.md:37]: s/container/container/

**Review by doismellburning — CHANGES_REQUESTED** [review]
There is quite a lot going on here that makes it more challenging to meaningfully review than I'd like; could you try to keep it to commit-per-logical-change please? Trying to follow what you're trying to do is hard when e.g. the README diff has everything from formatting changes to adding LXD instructions to moving to Make usage etc.

Splitting it up would not only make it easier for me to review, it will help when looking back at history, trying to understand intent, etc. - I'd appreciate it if you would please.

**Comment by glower** [comment]
> Splitting it up would not only make it easier for me to review, it will help when looking back at history, trying to understand intent, etc. - I'd appreciate it if you would please.

Sure, make sense. I will split the automation changes and README changes

File: run-checks [hunk 8]
```diff
@@ -13,11 +13,9 @@ set -eu
 if which goctest >/dev/null; then
     goctest="goctest"
 else
-    goctest="go test"
```
Comment (glower) [inline run-checks:16]: '-mod=vendor' tells go to use vendor code from the `vendor` folder

File: .github/workflows/nodejs.yml [hunk 9]
```diff
@@ -14,14 +14,7 @@ jobs:
         node-version: 12.x
     - name: Build frontend
       run: |
-        cd webapp-admin
```
Comment (glower) [inline .github/workflows/nodejs.yml:17]: this is moved to the Makefile

File: .travis.yml [hunk 10]
```diff
@@ -1,50 +1,26 @@
-sudo: required
```
Comment (glower) [inline .travis.yml:1]: this is all not needed, new config file is clean

File: .travis.yml [hunk 11]
```diff
@@ -1,50 +1,26 @@
...
       env:
         - NODE_ENV=test
-      before_install:
-        - cd webapp-admin; npm install
-      install:
-        - npm install -g codecov
       script:
-        - npm run test:ci
-      after_success:  
-        curl -s https://codecov.io/bash | bash /dev/stdin -f junit.xml
```
Comment (glower) [inline .travis.yml:50]: codecov never worked for the frontend tests, removed

**Comment by glower** [comment]
* New github actions are removed and will be introduced in the new PR
* Unnecessary changes in the README are removed
* Added missing Makefile targets for frontend tests and for code vendoring

Documentation about LXD container could me also moved to a separate PR but I think it make sense here too

File: .travis.yml [hunk 12]
```diff
@@ -1,50 +1,26 @@
...
-  - go get -u github.com/alexkohler/nakedret
-  - go get ./...
-
-install:
-  - echo $GOPATH
-  - echo "Remaining install is done by the test script."
-  - true
-
-script: sh -v ./run-checks $TEST_SUITE
+dist: focal
```
Comment (Roadmaster) [inline .travis.yml:1]: Note that it'd be best to test on the same distro where the service is deployed. That said, it's OK to leave this as focal and we should instead change the build/deploy process to use focal :) (tl;dr no change needed here).

**Review by Roadmaster — APPROVED** [review]
LGTM, a tiny comment but should not block the merge.

I agree with Kristian's request to split this into meaningful commits but if the resulting diff is the same, shouldn't need to request another review.

File: .travis.yml (hunk 12 above)
Comment (glower) [reply .travis.yml:1]: Yes, that makes sense. I will change the build/test process to use focal in the next PR

**Pull request merged**